### PR TITLE
Added solution for HW_14.2

### PR DESCRIPTION
**Pull request by polinakulyk**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/Application.java [hunk 1]
```diff
@@ -38,4 +54,95 @@ public static void main(String[] args) {
...
+        User user = new User().setName("Bucket Head");
+
+        OrderService orderService =
+                (OrderService) injector.getInstance(OrderService.class);
+        UserService userService =
+                (UserService) injector.getInstance(UserService.class);
+
+        userService.create(user);
+
+        Order o = orderService.completeOrder(new ArrayList<Product>() {{
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/Application.java:84]: List.of(...)

File: src/main/java/mate/academy/internetshop/dao/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public List<Order> getByUser(Long userId) {
+        return Storage.orders.stream()
+                .filter(o -> o.getUser().getId().equals(userId))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Order update(Order order) {
+        IntStream.range(0, Storage.orders.size())
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/dao/OrderDaoImpl.java:36]: Storage.orders.stream() .... 
iterate over collection

File: src/main/java/mate/academy/internetshop/dao/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public Optional<ShoppingCart> getByUser(Long userId) {
+        return Storage.shoppingCarts.stream()
+                .filter(sc -> sc.getUser().getId().equals(userId))
+                .findFirst();
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        IntStream.range(0, Storage.shoppingCarts.size())
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/dao/ShoppingCartDaoImpl.java:35]: same as for orders

File: src/main/java/mate/academy/internetshop/dao/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public Optional<User> get(Long id) {
+        return Storage.users.stream()
+                .filter(u -> u.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public User update(User user) {
+        IntStream.range(0, Storage.users.size())
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/dao/UserDaoImpl.java:28]: same as for orders

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public Order completeOrder(List<Product> products, User user) {
+        if (products.isEmpty()) {
+            throw new IllegalArgumentException("Product list empty");
+        }
+        if (userDao.get(user.getId()).isEmpty()) {
+            throw new IllegalArgumentException("User not found: " + user);
+        }
+        List<Product> allProducts = productDao.getAll();
+        products.stream().forEach(orderProduct -> {
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java:38]:
```suggestion
        products.forEach(orderProduct -> {
```

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            if (!allProducts.stream().anyMatch(existingProduct ->
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java:39]: use noneMatch

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                    existingProduct.getId().equals(orderProduct.getId()))) {
+                throw new IllegalArgumentException("Product not found: " + orderProduct);
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java:41]: try to use orElseThrow if applicable

File: src/main/java/mate/academy/internetshop/service/ShoppingCartServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        Optional<ShoppingCart> shoppingCartOpt = shoppingCartDao.get(shoppingCart.getId());
+        if (shoppingCartOpt.isEmpty()) {
+            return null;
+        }
+        shoppingCart = shoppingCartOpt.get();
+        if (!shoppingCart.getProducts().stream()
+                .anyMatch(p -> p.getId().equals(product.getId()))) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/internetshop/service/ShoppingCartServiceImpl.java:28]: try to use noneMatch

File: src/main/java/mate/academy/internetshop/Application.java (hunk 1 above)
Comment (polinakulyk) [reply src/main/java/mate/academy/internetshop/Application.java:84]: done

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java (hunk 5 above)
Comment (polinakulyk) [reply src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java:38]: done

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java (hunk 6 above)
Comment (polinakulyk) [reply src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java:39]: done

File: src/main/java/mate/academy/internetshop/service/ShoppingCartServiceImpl.java (hunk 8 above)
Comment (polinakulyk) [reply src/main/java/mate/academy/internetshop/service/ShoppingCartServiceImpl.java:28]: done

File: src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java (hunk 7 above)
Comment (polinakulyk) [reply src/main/java/mate/academy/internetshop/service/OrderServiceImpl.java]: attempted to)

**Pull request merged**